service/src/test/java/com/epam/esm/TagServiceImplTest.java

```diff
@@ -0,0 +1,116 @@
+package com.epam.esm;
+
+import com.epam.esm.dto.TagDTO;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.exception.EntityAlreadyExistsException;
+import com.epam.esm.exception.EntityNotFoundException;
+import com.epam.esm.repository.TagRepository;
+import com.epam.esm.repository.TagRepositoryImpl;
+import com.epam.esm.service.impl.TagServiceImpl;
+import com.epam.esm.specification.TagAllSpecification;
+import com.epam.esm.validator.TagDTOValidator;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```

Comment:
see comment for CertificateServiceImplTest

Reply:
Also see my comments for CertificateServiceImplTest :)